Week7/CatApi/app/src/main/java/com/example/catapi/ui/SecondFragment.kt

```diff
@@ -0,0 +1,195 @@
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import java.lang.reflect.Type
+
+
+/**
+ * A simple [Fragment] subclass as the second destination in the navigation.
+ */
+@RequiresApi(Build.VERSION_CODES.M)
+class SecondFragment : Fragment() {
+
+    private lateinit var breedMap : Map<String,String>
+    private var image = ""
+    private lateinit var model : MyViewModel
+
+
+    private val networkStatusChecker by lazy {
+        NetworkStatusChecker(activity?.getSystemService(ConnectivityManager::class.java))
+    }
+
+    // ------------------------------------ For Api with List<Breed> adapter
+    private val apiServiceBreed by lazy {
+        val type: Type = Types.newParameterizedType(List::class.java, Breed::class.java)
+        val parserM = Moshi.Builder().build()
+        val reportAdapter: JsonAdapter<List<Breed>> = parserM.adapter(type)
+        buildApiService(parserM)
+    }
+
+    private val remoteApiBreed by lazy {
+        RemoteApi(apiServiceBreed)
+    }
+    //----------------------------------------------------------------------
+
+    // ------------------------------------ For Api with List<Cat> adapter
+    private val apiServiceImage by lazy {
+
+        val type: Type = Types.newParameterizedType(List::class.java, Cat::class.java)
+        val parserM = Moshi.Builder().build()
+        val reportAdapter: JsonAdapter<List<Cat>> = parserM.adapter(type)
+        buildApiService(parserM)
+
+    }
+
+    private val remoteApiImage by lazy {
+        RemoteApi(apiServiceImage)
+    }
+    //----------------------------------------------------------------------
+
+    private val genreList = arrayOf("Male","Female")
+
+    private fun obtainBreedList(){
+        val loadingDialog = activity?.let { LoadingDialog(it) }
+        loadingDialog?.startDialog()
+
+        GlobalScope.launch(Dispatchers.Main) {
+            if(!networkStatusChecker.performIfConnectedToInternet {
+                val result = remoteApiBreed.getBreedList()
+
+                if (result is Success) {
+                    breedMap = result.data.map { it.name to it.id }.toMap()
+
+                    //-----------------------------------------------Display the breeds available in the spinner
+                    val breedAdapter = activity?.let {
+                        ArrayAdapter(
+                            it,
+                            android.R.layout.simple_spinner_dropdown_item,
+                            breedMap.keys.toTypedArray()
+                        )
+                    }
+                    breedAdapter?.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+                    sp_breed.adapter = breedAdapter
+                    loadingDialog?.dismissDialog()
+                    //------------------------------------------------------------------------------------------
+
+                }
+            }){
```

Comment:
I'm a little confused, perhaps there should be an else on this line? It seems as though the opening curly brace is not tied to the if statement above.